js/saveState.js

```diff
@@ -0,0 +1,42 @@
+import _ from "lodash";
+
+import { COOKIE_LAST_STATE, COOKIE_OMIT_FROM_STATE } from "./constants/AppConstants";
+
+// store the cookie here in case we're testing
+let lastCookie = "";
+
+function cookiesToMap (cookieStr) {
+  return _.chain(cookieStr.split(/;\s*/))
+          .map((keyValStr) => keyValStr.split(/=/))
+          .fromPairs()
+          .value();
+}
+
+export function restoreState() {
+	let cookie = lastCookie;
```

Comment:
Oh, weird! After investigating, it looks like the file has a mix of spaces and tabs, because I switched the mode in my editor (SublimeText) from "tabs with width of 2 spaces" to "2 spaces as tabs" midway through editing the file. I thought it would convert existing tabs but it didn't, and I didn't notice because the width was the same. I'll definitely fix this.